content/articles/protobuf-instead-of-json-in-frontend.md

```diff
@@ -0,0 +1,107 @@
+Title: Using Protobuf instead of JSON to communicate with a frontend
+Date: 2015-05-22
+Short_summary: Protobuf (or others tools like Thrift) are very often used to communicate between services. Let's see how it would work to use it between a server and a javascript frontend
+Category: Dev
+Authors: Vincent
+
+*Reading time: ~15 minutes.*  
+
+Protobuf (or other binary serialization formats like Thrift) are widely used to communicate between services. JSON is overwhelmingly used for backend <-> frontend communication and for APIs.  
+Let's see how a client/server using Protobuf would look like.
+<!-- PELICAN_END_SUMMARY --> 
+
+## What is Protobuf
+[Protobuf](https://developers.google.com/protocol-buffers/docs/overview) is a binary serialization format from Google meant to serialize structured data. It has clients in most languages, including Python and Javascript which we are going to use in our toy app.  
+
+Tools like Protobuf have a few advantages over JSON:
+
+- smaller in size
+- typed
+- offer a common interface for all your services, you can just update your .proto or .thrift and share those with the services using it, as long as there is a client for that language
+
+[Thrift](https://thrift.apache.org/) does more as it also allows you to define methods as well as structs.
+
+## Example app
+As an example we are going to make a small address book app. Coincidentally that's the example that the Python Protobuf doc is using as well but that wasn't on purpose I swear! The data is generated randomly on startup of the flask app.  
```

Comment:
I would spell doc and app in full and maybe add some punctuation like "that wasn't on purpose, I swear!"